drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -1919,15 +1933,37 @@ function DragAndDropBlock(runtime, element, configuration) {
     var isPastDue = function () {
         return !configuration.has_deadline_passed;
     };
-
-    var canShowAnswer = function() {
-        return configuration.mode === DragAndDropBlock.ASSESSMENT_MODE && !attemptsRemain();
+    var isAttempted = function () {
+        return state.attempts > 0;
+    };
+    var isCorrect = function () {
+        return state.last_action_correct;
+    };
+    var isClosed = function () {
+        if(isPastDue() || !attemptsRemain()) return true;
+        return false;
     };
 
     var attemptsRemain = function() {
         return !configuration.max_attempts || configuration.max_attempts > state.attempts;
     };
 
+    var canShowAnswer = function() {
+        if(configuration.mode !== DragAndDropBlock.ASSESSMENT_MODE) return false;
+        else if (configuration.showanswer === SHOW_ANSWER_STATUSES.NEVER) return false;
+        else if (configuration.showanswer === SHOW_ANSWER_STATUSES.ATTEMPTED) return isAttempted() || isPastDue();
```

Comment:
@abodacs Setting the "Show Answer" value to "attempted" is causing the the button to be enabled even before I have made the first attempt. I think the issue is with the use of `isPastDue` function here. Since `isPastDue` is actually returning the *inverse* of `configuration.has_deadline_passed`, the logic of the Python equivalent `self.is_attempted or self.has_submission_deadline_passed` is being altered, resulting in wrong results.

Reply:
@tecoholic 
Thanks for detailed answered

I removed the inverse from ```isPastDue``` and edited ```canSubmitAttempt``` function based on the inverse

``` javascript
var canSubmitAttempt = function() {
        return Object.keys(state.items).length > 0 && !isPastDue() && attemptsRemain() && !submittingLocation();
    };

```

Reply:
@abodacs Great. I tested the various settings again. They seem to work as expected :)